sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -0,0 +1,101 @@
+## API Report File for "@azure/keyvault-admin"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { TokenCredential } from '@azure/core-http';
+
+// @public
+export interface AccessControlClientOptions extends coreHttp.PipelineOptions {
+    serviceVersion?: SUPPORTED_API_VERSIONS;
+}
+
+// @public
+export interface CreateRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface DeleteRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface GetRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export class KeyVaultAccessControlClient {
+    constructor(vaultUrl: string, credential: TokenCredential, pipelineOptions?: AccessControlClientOptions);
+    createRoleAssignment(roleScope: RoleAssignmentScope, name: string, roleDefinitionId: string, principalId: string, options?: CreateRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    deleteRoleAssignment(roleScope: RoleAssignmentScope, name: string, options?: DeleteRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    getRoleAssignment(roleScope: RoleAssignmentScope, name: string, options?: GetRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    listRoleAssignments(roleScope: RoleAssignmentScope, options?: ListRoleAssignmentsOptions): PagedAsyncIterableIterator<KeyVaultRoleAssignment>;
+    listRoleDefinitions(roleScope: RoleAssignmentScope, options?: ListRoleDefinitionsOptions): PagedAsyncIterableIterator<KeyVaultRoleDefinition>;
+    readonly vaultUrl: string;
+}
+
+// @public
+export interface KeyVaultPermission {
+    actions?: string[];
+    dataActions?: string[];
+    notActions?: string[];
+    notDataActions?: string[];
+}
+
+// @public
+export interface KeyVaultRoleAssignment {
+    readonly id?: string;
+    readonly name?: string;
+    properties?: KeyVaultRoleAssignmentPropertiesWithScope;
+    readonly type?: string;
+}
+
+// @public
+export interface KeyVaultRoleAssignmentProperties {
+    principalId?: string;
+    roleDefinitionId?: string;
+}
+
+// @public
+export interface KeyVaultRoleAssignmentPropertiesWithScope extends KeyVaultRoleAssignmentProperties {
+    scope?: RoleAssignmentScope;
+}
+
+// @public
+export interface KeyVaultRoleDefinition {
+    assignableScopes?: string[];
+    description?: string;
+    readonly id?: string;
+    readonly name?: string;
+    permissions?: KeyVaultPermission[];
+    roleName?: string;
+    roleType?: string;
+    readonly type?: string;
```

Comment:
having a `type` and a `roleType` feels a little confusing to me, especially since `type` is so generic

Reply:
This is something we would need to change across languages too. I'll ask them about this today.

Reply:
After recent feedback from the .Net team, through Teams, we agreed on enforcing most of the properties. So, please review again!